clippy_lints/src/collapsible_if.rs

```diff
@@ -209,6 +211,71 @@ impl CollapsibleIf {
         !matches!(cond.kind, ExprKind::Let(..))
             || (cx.tcx.sess.edition().at_least_rust_2024() && self.msrv.meets(cx, msrvs::LET_CHAINS))
     }
+
+    fn check_collapsible_if_nested_if_else(&self, cx: &LateContext<'_>, if_expr: &Expr<'_>) {
+        if let Some(If {
+            cond,
+            then: Expr {
+                kind: ExprKind::Block(then_block, ..),
+                ..
+            },
+            r#else: Some(else_expr),
+        }) = If::hir(if_expr)
+            // && let Some(Expr {
+            //     kind: ExprKind::If(inner_cond, inner_then, ..),
+            //     ..
+            // }) = expr_block(then_block)
+            && let Some(inner_if_expr) = expr_block(then_block)
+            && let Expr {
+                kind: ExprKind::If(inner_cond, inner_then, ..),
+                ..
+            } = inner_if_expr
+            && !block_starts_with_significant_tokens(cx, then_block, inner_if_expr, self.lint_commented_code)
+            && SpanlessEq::new(cx).eq_expr(inner_then, else_expr)
+        {
+            let first_cond_sugg = match cond.kind {
+                ExprKind::Binary(bin_op, lhs, rhs) => {
+                    let new_bin_op = if let BinOpKind::Ne = bin_op.node {
+                        BinOpKind::Eq
+                    } else {
+                        BinOpKind::Ne
+                    };
+
+                    make_binop(
+                        new_bin_op,
+                        &Sugg::hir_with_applicability(cx, lhs, "..", &mut Applicability::HasPlaceholders),
```

Comment:
This is not the way the last argument is supposed to be used. You have to create a variable initialized to `Applicability::MachineApplicable` (for example named `app`) and pass it to every call to `Sugg::*()` or `snippet_*()` so that it can be updated, and then you can use it later when giving the suggestion.